test/e2e/sagemaker/service_bootstrap.py

```diff
@@ -76,10 +76,14 @@ def create_data_bucket() -> str:
     return bucket_name
 
 
+"""ExecutionRoleARN needs to be created first so the role is ready to use
+before the test starts. Hence it should be the first thing to create in
+bootstrapping
+"""
 def service_bootstrap() -> dict:
     logging.getLogger().setLevel(logging.INFO)
 
     return TestBootstrapResources(
-        create_data_bucket(),
         create_execution_role(),
+        create_data_bucket(),
```

Comment:
`create_data_bucket()` does not rely on `create_execution_role()` - why does this order matter?

Reply:
I noticed is that all tests including training were failing due to role not being available when the tests ran. But they eventually succeed after the role became available. My observation is that it takes some time for the role creation and and the test starts before the role is available. Tests error out saying that role is not present. So, creating the role first gives it time to become available for the test and resolved the issue for me. The error is noticeable in the Batch Transform test because the model creation fails from the boto3 call and with this change, model creation call succeeds.

Reply:
Is it not sufficient to wait for the role to become available as part of the bootstrapping process?

Reply:
I changed the test to use sleep and added a comment indicating that it is to work around this issue. For now, sleep is sufficient for the role to become available